x-pack/legacy/plugins/apm/server/lib/index_pattern/getKueryBarIndexPattern.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+import { StaticIndexPattern } from 'ui/index_patterns';
+import { IndexPatternsService } from '../../../../../../../src/legacy/server/index_patterns/service';
+import { Setup } from '../helpers/setup_request';
+
+export const getKueryBarIndexPattern = async ({
+  request,
+  processorEvent,
+  setup
+}: {
+  request: Legacy.Request;
+  processorEvent?: string;
+  setup: Setup;
+}) => {
+  const { config } = setup;
+
+  const indexPatternsService = new IndexPatternsService(
+    (...args: [any, any, any]) =>
```

Comment:
I think typing these out will make the signature easier to read:
```ts
(endpoint: string, params?: Record<string, any>, options?: CallAPIOptions) =>
```

Reply:
Agree that [any,any,any] doesn't really help us. I've made a slight change based on your suggestion, because I think it should be explicit that this function is simply middleware and does not really care what arguments are passed to it:

```typescript
const indexPatternsService = new IndexPatternsService(
    (...rest: Parameters<APICaller>) =>
      request.server.plugins.elasticsearch
        .getCluster('data')
        .callWithRequest(request, ...rest)
  );
````

Reply:
Thanks 👍